File: cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{   
    float dollars = 0; //initializes variable dollars
    do
    {
        dollars = get_float("How much change is owed?\n"); //asks human for change owed
    }
    while (dollars <= 0); //ensures amount is positive
    
    int change = round(dollars * 100); //rounds dollars and turns it into cents (i.e. 1.40 = 140)
    
    int coins = 0; //initializes coin counter
    
    for (; change >= 25; change = change - 25)  //calculates how many quarters can be used
    {
        coins = coins + 1;
    }
    
    for (; change >= 10; change = change - 10)  //calculates how many dimes can be used
    {
        coins = coins + 1;
    }
    
    for (; change >= 5; change = change - 5)  //calculates how many nickels can be used
    {
        coins = coins + 1;
    }
    
    for (; change > 0; change = change - 1)  //calculates how many pennies can be used
    {
        coins = coins + 1;
    }
    
    printf("Minimum number of coins:\n");
    printf("%i\n", coins); //prints output
}

File: cash.py
from cs50 import get_float

while True:
    dollars = get_float("Change owed: ")
    if dollars > 0:  # runs until dollars is a positive number
        break

change = round(dollars * 100)  # turns dollars into cents
coins = 0  # initializes counter to 0

while True:
    if change >= 25:
        change -= 25
        coins += 1
    if change < 25:
        break  # runs until cannot use any more quarters
while True:
    if change >= 10:
        change -= 10
        coins += 1
    if change < 10:
        break  # runs until cannot use any more dimes
while True:
    if change >= 5:
        change -= 5
        coins += 1
    if change < 5:
        break  # runs until cannot use any more nickels
while True:
    if change >= 1:
        change -= 1
        coins += 1
    if change == 0:
        break  # runs until cannot use any more pennies

print(coins)